de/ODE.py

```diff
@@ -3,44 +3,56 @@
 '''
 import numpy as np
 from scipy.integrate import solve_ivp
-import autograd.numpy as anp
-from autograd import jacobian
-from numba import njit
+from os.path import join, dirname
+from julia.api import Julia
+jl = Julia(compiled_modules=False)
 
-def solver(n_x, N, x0, N_t, p, beta):
+from julia import Base
+
+
+path_here = dirname(dirname(__file__))
+Base.MainInclude.include(join(path_here, "de/model.jl"))
+
+def julia_solver(ps):
     '''
-    Function receives time series and a set of parameters,
-       then return the simulation of ODE.
+    Function receives a set of parameters,
+       then return the simulation of ODE over time.
        p = [eps, w, alpha] as 1D array
     '''
-    eps = p[0:n_x]
-    w = p[n_x:(n_x ** 2 + n_x)].reshape((n_x, n_x))
-    alpha = p[(n_x ** 2 + n_x):]
-    sol = np.ones((N, n_x))
-    t = np.arange(N_t)
-    for i in range(N):
-        #sol[i, :] = np.transpose(solve_ivp(ODE, (0, N_t), x0, args=(eps, w, alpha, beta[:, i]), t_eval=[N_t], jac=jacobian_autograd, method="LSODA").y)
-        sol[i, :] = np.transpose(solve_ivp(ODE, (0, N_t), x0, args=(eps, w, alpha, beta[:, i]), t_eval=[N_t], method="LSODA").y)
-    return sol
+    return jl.eval('solveODE')(ps)
+
+def julia_sol_matrix(ps):
+    '''
+    Function receives a set of parameters then returns the simulation of ODE at the last timepoint for all KO models,
+        mimicking the experimental data.
+        p = [eps, w, alpha] as 1D array
+    '''
+    return jl.eval('sol_matrix')(ps)
 
-@njit
-def ODE(t, y, eps, w, alpha, beta):
-    '''The ODE system:
-    Parameters = eps: Value that bound the saturation effect
-                 w: Interaction between components
-                 alpha: Degradation rate
-                 beta: Knock-out effects
+def reshape_params(ps):
+    '''
+    Function receives a set of parameters then returns eps, w, and alpha as separate arrays
+    p = [eps, w, alpha] as 1D array
     '''
-    return eps * (1 + np.tanh(np.dot(w, y))) - (alpha*beta) * y
+    eps = ps[0:83]
+    w = ps[83:(83 ** 2 + 83)].reshape((83, 83))
+    alpha = ps[(83 ** 2 + 83):]
+    return eps, w, alpha
 
-def jacobian_autograd(t, y, eps, w, alpha, beta):
+def ODE(t, y, eps, w, alpha):
     '''
-    Given a set of parameters and the state of system, it will return the Jacobian of the system.
+    Function receives a set of parameters,
+       then return the simulation of ODE over time.
     '''
-    return jacobian(ODE_anp)(y, eps, w, alpha, beta)
+    return eps * (1 + np.tanh(np.dot(w, y))) - alpha * y
 
-def ODE_anp(y, eps, w, alpha, beta):
+def solver(x0, N_t, p):
```

Comment:
Why keep the Python solver? It's better to not duplicate functionality in two places, because the two implementations could potentially be different in some way.